### PR TITLE
Signup Route

### PR DESCRIPTION
**Pull request by AbdallahAbujazar**
Change fields names in User Model like full_name to fullName Relates #18

### CODE REVIEW

File: src/controllers/auth/signup.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      password,
+      phone,
+      userImage,
+      profileImage,
+      address,
+      userType,
+      status
+    }: IUser = req.body
+
+    await validateSignup(req.body)
```
Comment (Mu7ammadAbed) [inline src/controllers/auth/signup.ts:47]:
```suggestion
    }: IUser = await validateSignup(req.body)
```

File: src/controllers/auth/signup.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import { type Request, type Response } from 'express'
+import jwt, { type Secret } from 'jsonwebtoken'
+import dotenv from 'dotenv'
+import bcrypt from 'bcrypt'
+import User from '../../models/User'
+import { createUser } from '../../queries/user/signup'
+import { type IUser } from '../../interfaces/models'
```
Comment (Mu7ammadAbed) [inline src/controllers/auth/signup.ts:7]:
```suggestion
import { type IUser } from '../../interfaces/fakeDataTypes'
```

File: src/validation/auth/signup.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import * as Yup from 'yup'
+import { type IUser } from '../../interfaces/models'
```
Comment (Mu7ammadAbed) [inline src/validation/auth/signup.ts:2]:
```suggestion
import { type User as IUser } from '../../interfaces/fakeDataTypes'
```

File: src/validation/auth/signup.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    address: Yup.string(),
+    userType: Yup.string()
+      .required('User Type is required')
+      .test('is-valid-user-type', 'Invalid User Type', (value) => value === 'admin' || value === 'user'),
+    status: Yup.string()
+      .required('Status is required')
+      .test('is-valid-status', 'Invalid Status', (value) => value === 'active' || value === 'inactive')
+  })
+
+  await schema.validate(data, { abortEarly: false })
```
Comment (Mu7ammadAbed) [inline src/validation/auth/signup.ts:21]:
```suggestion
  const result = await schema.validate(data, { abortEarly: false })
  return result
```

File: src/queries/user/signup.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  fullName: string
+  email: string
+  password: string
+  phone: string
+  userImage: string
+  profileImage: string
+  address: string
+  userType: string
+  status: string
+}
```
Comment (Mu7ammadAbed) [inline src/queries/user/signup.ts:14]:
```suggestion
interface CreateUserProps {
  fullName: string
  email: string
  password: string
  phone?: string
  userImage?: string
  profileImage?: string
  address?: string
  userType?: string
  status: string
}

interface INewUser extends IUser {
  userId: number
}
```

File: src/controllers/auth/signup.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+      status
+    }: IUser = req.body
+
+    await validateSignup(req.body)
+
+    const existingUser = await User.findOne({ where: { email } })
+    if (existingUser != null) {
+      res.status(400).json({ error: 'Email already exists' })
+      return
+    }
```
Comment (Mu7ammadAbed) [inline src/controllers/auth/signup.ts:53]: This query should be moved to queries folder

File: src/queries/user/signup.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  fullName,
+  email,
+  password,
+  phone,
+  userImage,
+  profileImage,
+  address,
+  userType,
+  status
+}: CreateUserProps): Promise<IUser> => {
```
Comment (Mu7ammadAbed) [inline src/queries/user/signup.ts:26]:
```suggestion
}: CreateUserProps): Promise<INewUser > => {
```

File: src/controllers/auth/signup.ts [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        fullName,
+        email: newUser.email,
+        phone: newUser.phone,
+        userImage,
+        profileImage,
+        address,
+        userType,
+        status
+      }
+    })
```
Comment (Mu7ammadAbed) [inline src/controllers/auth/signup.ts:85]:
```suggestion
res.cookie('token', token, { httpOnly: true, secure: process.env.NODE_ENV === 'production' })
      .json({
        message: 'User Created Successfully',
        data: {
          fullName,
          email: newUser.email,
          phone: newUser.phone,
          userImage,
          profileImage,
          address,
          userType,
          status
        }
      })
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: src/interfaces/models.ts [hunk 9]
```diff
@@ -58,15 +58,15 @@ interface IProduct extends Model<InferAttributes<IProduct>, InferCreationAttribu
...
   email: string
   password: string
-  user_image: CreationOptional<string>
-  profile_image: CreationOptional<string>
+  userImage: CreationOptional<string>
+  profileImage: CreationOptional<string>
   address: CreationOptional<string>
   phone: CreationOptional<string>
-  user_type: string
+  userType: string
```
Comment (Mu7ammadAbed) [inline src/interfaces/models.ts:69]:
```suggestion
  userType?: string
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: src/controllers/auth/signup.ts (hunk 8 above)
Comment (Mu7ammadAbed) [reply src/controllers/auth/signup.ts]: @AbdallahAbujazar You forgot this one

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by MohammedSallout — APPROVED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by haitham-akram — APPROVED** [review]

**Pull request merged**